docs/retrieval_augmented_generation/rag_cookbook.ipynb

```diff
@@ -0,0 +1,545 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# RAG Cookbook\n",
+    "\n",
+    "In this notebook, we show the useage of CAMEL `RetrievalModule` in both customized way and default auto way. We will also show how to combine `RetrievalModule` with `ChatAgent`, and further combine with `RolePlaying` by using `Function Calling`.\n",
+    "\n",
+    "4 main parts included:\n",
+    "- Customized RAG\n",
+    "- Default RAG\n",
+    "- Single Agent with RAG\n",
+    "- Role-playing with RAG"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load Data\n",
+    "\n",
+    "Let's first load the CAMEL paper from https://arxiv.org/pdf/2303.17760.pdf. This will be our local example data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir -p 'local_data/'\n",
+    "!wget --user-agent \"Mozilla\" \"https://arxiv.org/pdf/2303.17760.pdf\" -O \"local_data/camel paper.pdf\""
+   ]
```

Comment:
I'm using **Windows** which doesn't actually succeed when I run this cell. `mkdir -p` doesn't seem to work on Windows either, and a special download is required if we want to use` wget`. Maybe it could be done with python code? 

ex:
```python
import os
import requests

os.makedirs('local_data', exist_ok=True)

url = "https://arxiv.org/pdf/2303.17760.pdf"
response = requests.get(url)
with open('local_data/camel paper.pdf', 'wb') as file:
     file.write(response.content)

Reply:
Thanks for the valuable input, `wget ` may not be available in all environments, will update the code with `request` method as you suggested.